### PR TITLE
KAFKA-16683: Extract security-related helpers from scala.TestUtils to java class

### PR DESCRIPTION
**Pull request by FrankYang0529**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@FrankYang0529 please add jira number

**Comment by chia7712** [comment]
@FrankYang0529 Could you please rebase code?

**Comment by FrankYang0529** [comment]
Hi @chia7712, I rebased the code and all failed cases can pass on my laptop. Could you take a look when you have time? Thank you.

File: core/src/test/java/kafka/security/JaasTestUtils.java [hunk 1]
```diff
@@ -269,4 +275,108 @@ private static void writeToFile(File file, List<JaasSection> jaasSections) throw
...
+
+    public static Properties sslConfigs(ConnectionMode mode,
+                                        boolean clientCert,
+                                        Optional<File> trustStoreFile,
+                                        String certAlias,
+                                        String certCn,
+                                        String tlsProtocol) throws Exception {
+        File trustStore = trustStoreFile.orElseThrow(() -> new Exception("SSL enabled but no trustStoreFile provided"));
+
+        TestSslUtils.SslConfigsBuilder sslConfigsBuilder = new TestSslUtils.SslConfigsBuilder(mode)
```
Comment (chia7712) [inline core/src/test/java/kafka/security/JaasTestUtils.java]:
```java
        File trustStore = trustStoreFile.orElseThrow(() -> new Exception("SSL enabled but no trustStoreFile provided"));
        Properties sslProps = new Properties();
        sslProps.putAll(new TestSslUtils.SslConfigsBuilder(mode)
                .useClientCert(clientCert)
                .createNewTrustStore(trustStore)
                .certAlias(certAlias)
                .cn(certCn)
                .tlsProtocol(tlsProtocol)
                .build());
        return sslProps;
```

File: core/src/test/java/kafka/security/JaasTestUtils.java [hunk 2]
```diff
@@ -269,4 +275,108 @@ private static void writeToFile(File file, List<JaasSection> jaasSections) throw
...
+     * Returns security configuration options for broker or clients
+     *
+     * @param connectionMode Client or server mode
+     * @param securityProtocol Security protocol which indicates if SASL or SSL or both configs are included
+     * @param trustStoreFile Trust store file must be provided for SSL and SASL_SSL
+     * @param certAlias Alias of certificate in SSL key store
+     * @param certCn CN for certificate
+     * @param saslProperties SASL configs if security protocol is SASL_SSL or SASL_PLAINTEXT
+     * @param tlsProtocol TLS version
+     * @param needsClientCert If not empty, a flag which indicates if client certificates are required. By default
```
Comment (chia7712) [inline core/src/test/java/kafka/security/JaasTestUtils.java]: `By default,`

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 thanks for your patch. some trivial comments are left. PTAL

**Comment by FrankYang0529** [comment]
Hi @chia7712, thanks for the suggestion. I addressed all comments.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**